stencil/src/components/pulumi-api-doc-filterable-nav/pulumi-api-doc-filterable-nav.tsx

```diff
@@ -31,19 +31,34 @@ export class PulumiApiDocFilterableNav {
     }
 
     componentWillLoad() {
-        // Parse the JSON that's passed through as a prop.
-        this.parsedNodes = JSON.parse(this.nodes);
-        // Before the user interacts with the filter, the nodes to render should be the full nav tree.
-        this.currentlyRenderedNodes = this.parsedNodes;
+        this.loadNavigationData();
     }
 
-    private textInput?: HTMLInputElement;
+    async loadNavigationData() {
+        this.isLoading = true;
+
+        try {
+            const response = await fetch(`/registry/packages/navs/${this.packageName}.json`);
+            const data = await response.json();
+            this.parsedNodes = data;
+
+            // Before the user interacts with the filter, the nodes to render should be the full nav tree.
+            this.currentlyRenderedNodes = this.parsedNodes;
+        } catch (error) {
+            return;
+        }
+
+        this.isLoading = false;
```

Comment:
One thing that we could do to buy ourselves some time is to defer-update this with a timeout of 0 (`setTimeout(() => this.isLoading = false, 0)`). I am curious to see if that would help with ensuring that the browser isn't trying to render the tree as well as making them visible because `loading` is set to `false`.